File: Polygonal numbers.py
"""
Finding the first 100 polygonal numbers with sides between 3 and 10000


~~Documentation~~
https://en.wikipedia.org/wiki/Polygonal_number

~~Links~~
GitHub: https://github.com/DanielNicolae
Google Play: https://play.google.com/store/apps/developer?id=Don%C8%9Bu+Daniel+Nicolae
YouTube: https://www.youtube.com/channel/UCdxZEh4oSEsxDRc8MsTawjw?view_as=subscriber
GrabCad: https://grabcad.com/don.u.daniel.nicolae-1
WordPress: https://daniel25site.wordpress.com/
DeviantArt: http://free-gamer4ever.deviantart.com/

~~Author~~
Donțu Daniel Nicolae 2019
"""

def Triangular_numbers():
    for i in range(1,101):
        number = (i*i+i)/2
        nums=[]
        nums.insert(i,int(number))
        print (nums)
    
def Square_numbers():
    for i in range(1,101):
        number = i*i
        nums=[]
        nums.insert(i,int(number))
        print (nums)

def Pentagonal_numbers():
    for i in range(1,101):
        number = (3*i*i-i)/2
        nums=[]
        nums.insert(i,int(number))
        print (nums)

def Hexagonal_numbers():
    for i in range(1,101):
        number = (4*i*i-2*i)/2
        nums=[]
        nums.insert(i,int(number))
        print (nums)

def Heptagonal_numbers():
    for i in range(1,101):
        number = (5*i*i-3*i)/2
        nums=[]
        nums.insert(i,int(number))
        print (nums)

def Octagonal_numbers():
    for i in range(1,101):
        number = (6*i*i-4*i)/2
        nums=[]
        nums.insert(i,int(number))
        print (nums)

def Nonagonal_numbers():
    for i in range(1,101):
        number = (7*i*i-5*i)/2
        nums=[]
        nums.insert(i,int(number))
        print (nums)

def Decagonal_numbers():
    for i in range(1,101):
        number = (8*i*i-6*i)/2
        nums=[]
        nums.insert(i,int(number))
        print (nums)

def Hendecagonal_numbers():
    for i in range(1,101):
        number = (9*i*i-7*i)/2
        nums=[]
        nums.insert(i,int(number))
        print (nums)

def Dodecagonal_numbers():
    for i in range(1,101):
        number = (10*i*i-8*i)/2
        nums=[]
        nums.insert(i,int(number))
        print (nums)

def Tridecagonal_numbers():
    for i in range(1,101):
        number = (11*i*i-9*i)/2
        nums=[]
        nums.insert(i,int(number))
        print (nums)

def Tetradecagonal_numbers():
    for i in range(1,101):
        number = (12*i*i-10*i)/2
        nums=[]
        nums.insert(i,int(number))
        print (nums)

def Pentadecagonal_numbers():
    for i in range(1,101):
        number = (13*i*i-11*i)/2
        nums=[]
        nums.insert(i,int(number))
        print (nums)

def Hexadecagonal_numbers():
    for i in range(1,101):
        number = (14*i*i-12*i)/2
        nums=[]
        nums.insert(i,int(number))
        print (nums)

def Heptadecagonal_numbers():
    for i in range(1,101):
        number = (15*i*i-13*i)/2
        nums=[]
        nums.insert(i,int(number))
        print (nums)

def Octadecagonal_numbers():
    for i in range(1,101):
        number = (16*i*i-14*i)/2
        nums=[]
        nums.insert(i,int(number))
        print (nums)

def Enneadecagonal_numbers():
    for i in range(1,101):
        number = (17*i*i-15*i)/2
        nums=[]
        nums.insert(i,int(number))
        print (nums)

def Icosagonal_numbers():
    for i in range(1,101):
        number = (18*i*i-16*i)/2
        nums=[]
        nums.insert(i,int(number))
        print (nums)

def Icosihenagonal_numbers():
    for i in range(1,101):
        number = (19*i*i-17*i)/2
        nums=[]
        nums.insert(i,int(number))
        print (nums)

def Icosidigonal_numbers():
    for i in range(1,101):
        number = (20*i*i-18*i)/2
        nums=[]
        nums.insert(i,int(number))
        print (nums)

def Icositrigonal_numbers():
    for i in range(1,101):
        number = (21*i*i-19*i)/2
        nums=[]
        nums.insert(i,int(number))
        print (nums)

def Icositetragonal_numbers():
    for i in range(1,101):
        number = (22*i*i-20*i)/2
        nums=[]
        nums.insert(i,int(number))
        print (nums)

def Polygonal_numbers():
    sides=int(input('Choose the number of sides for the polygon (between 4 and 10000).\n'))
    if(sides<3 or sides>10000):
        sides=int(input('Insert a number between 4 and 10000.\n'))
    print("The first 100 polygonal numbers with "+str(sides)+" sides are:")
    for i in range(1,101):
        number = ((sides-2)*i*i-(sides-4)*i)/2
        nums=[]
        nums.insert(i,int(number))
        print (nums)
        
print('Enter a number between 3 and 24 to calculate the first 100 polygonal numbers that have between 3 and 24 sides.\n')
number=input('For polygonal numbers with sides greater than 24, please enter the number 25.\n')
a=int(number)
if a==3:
    print("The first 100 triangular numbers are:")
    Triangular_numbers()
elif a==4:
    print("The first 100 square numbers are:")
    Square_numbers()
elif a==5:
    print("The first 100 pentagonal numbers are:")
    Pentagonal_numbers()
elif a==6:
    print("The first 100 hexagonal numbers are:")
    Hexagonal_numbers()
elif a==7:
    print("The first 100 heptagonal numbers are:")
    Heptagonal_numbers()
elif a==8:
    print("The first 100 octagonal numbers are:")
    Octagonal_numbers()
elif a==9:
    print("The first 100 nonagonal numbers are:")
    Nonagonal_numbers()
elif a==10:
    print("The first 100 decagonal numbers are:")
    Decagonal_numbers()
elif a==11:
    print("The first 100 hendecagonal numbers are:")
    Hendecagonal_numbers()
elif a==12:
    print("The first 100 dodecagonal numbers are:")
    Dodecagonal_numbers()
elif a==13:
    print("The first 100 tridecagonal numbers are:")
    Tridecagonal_numbers()
elif a==14:
    print("The first 100 tetradecagonal numbers are:")
    Tetradegonal_numbers()
elif a==15:
    print("The first 100 pentadecagonal numbers are:")
    Pentadecagonal_numbers()
elif a==16:
    print("The first 100 hexadecagonal numbers are:")
    Hexadecagonal_numbers()
elif a==17:
    print("The first 100 heptadecagonal numbers are:")
    Heptadecagonal_numbers()
elif a==18:
    print("The first 100 octadecagonal numbers are:")
    Octadecagonal_numbers()
elif a==19:
    print("The first 100 enneadecagonal numbers are:")
    Enneadecagonal_numbers()
elif a==20:
    print("The first 100 icosagonal numbers are:")
    Icosagonal_numbers()
elif a==21:
    print("The first 100 icosihenagonal numbers are:")
    Icosihenagonal_numbers()
elif a==22:
    print("The first 100 icosidigonal numbers are:")
    Icosidigonal_numbers()
elif a==23:
    print("The first 100 icositrigonal numbers are:")
    Icositrigonal_numbers()
elif a==24:
    print("The first 100 icositetragonal numbers are:")
    Icositetragonal_numbers()
elif a==25:
    Polygonal_numbers()

File: Polygonal numbers.cpp
/*
Finding the first 100 polygonal numbers with sides between 3 and 10000


~~Documentation~~
https://en.wikipedia.org/wiki/Polygonal_number

~~Links~~
GitHub: https://github.com/DanielNicolae
Google Play: https://play.google.com/store/apps/developer?id=Don%C8%9Bu+Daniel+Nicolae
YouTube: https://www.youtube.com/channel/UCdxZEh4oSEsxDRc8MsTawjw?view_as=subscriber
GrabCad: https://grabcad.com/don.u.daniel.nicolae-1
WordPress: https://daniel25site.wordpress.com/
DeviantArt: http://free-gamer4ever.deviantart.com/

~~Author~~
Donțu Daniel Nicolae 2019
*/

#include <stdlib.h>
#include <iostream>
using namespace std;

int* Triangular_numbers();
int* Square_numbers();
int* Pentagonal_numbers();
int* Hexagonal_numbers();
int* Heptagonal_numbers();
int* Octagonal_numbers();
int* Nonagonal_numbers();
int* Decagonal_numbers();
int* Hendecagonal_numbers();
int* Dodecagonal_numbers();
int* Tridecagonal_numbers();
int* Tetradecagonal_numbers();
int* Pentadecagonal_numbers();
int* Hexadecagonal_numbers();
int* Heptadecagonal_numbers();
int* Octadecagonal_numbers();
int* Enneadecagonal_numbers();
int* Icosagonal_numbers();
int* Icosihenagonal_numbers();
int* Icosidigonal_numbers();
int* Icositrigonal_numbers();
int* Icositetragonal_numbers();

int* Polygonal_numbers();

int main()
{
	int polygonOfSides; 
	int* p;
	int i;
	
	cout << "Enter a number between 3 and 24 to calculate the first 100 polygonal numbers that have between 3 and 24 sides.\n";
	cout << "For polygonal numbers with sides greater than 24, please enter the number 25.\n";
	cin >> polygonOfSides;
	
	switch (polygonOfSides)
	{
	case 3:
		p = Triangular_numbers();
		
		cout << "The triangular numbers between 1 and 100:" << endl;
		for (i=1;i<=100;i++)
		{
			cout << p[i] << " ";
		}
		break;
	case 4:
		p = Square_numbers();
		cout << "The first 100 square numbers are:" << endl;
		for (i=1;i<=100;i++)
		{
			cout << p[i] << " ";
		}
		break;
	case 5:
		p = Pentagonal_numbers();
		cout << "The first 100 pentagonal numbers are:" << endl;
		for (i = 1;i <= 100;i++)
		{
			cout << p[i] << " ";
		}
		break;
	case 6:
		p = Hexagonal_numbers();
		cout << "The first 100 hexagonal numbers are:" << endl;
		for (i = 1;i <= 100;i++)
		{
			cout << p[i] << " ";
		}
		break;
	case 7:
		p = Heptagonal_numbers();
		cout << "The first 100 heptagonal numbers are:" << endl;
		for (i = 1;i <= 100;i++)
		{
			cout << p[i] << " ";
		}
		break;
	case 8:
		p = Octagonal_numbers();
		cout << "The first 100 octagonal numbers are:" << endl;
		for (i = 1;i <= 100;i++)
		{
			cout << p[i] << " ";
		}
		break;
	case 9:
		p = Nonagonal_numbers();
		cout << "The first 100 nonagonal numbers are:" << endl;
		for (i = 1;i <= 100;i++)
		{
			cout << p[i] << " ";
		}
		break;
	case 10:
		p = Decagonal_numbers();
		cout << "The first 100 decagonal numbers are:" << endl;
		for (i = 1;i <= 100;i++)
		{
			cout << p[i] << " ";
		}
		break;
	case 11:
		p = Hendecagonal_numbers();
		cout << "The first 100 hendecagonal numbers are:" << endl;
		for (i = 1;i <= 100;i++)
		{
			cout << p[i] << " ";
		}
		break;
	case 12:
		p = Dodecagonal_numbers();
		cout << "The first 100 dodecagonal numbers are:" << endl;
		for (i = 1;i <= 100;i++)
		{
			cout << p[i] << " ";
		}
		break;
	case 13:
		p = Tridecagonal_numbers();
		cout << "The first 100 tridecagonal numbers are:" << endl;
		for (i = 1;i <= 100;i++)
		{
			cout << p[i] << " ";
		}
		break;
	case 14:
		p = Tetradecagonal_numbers();
		cout << "The first 100 tetradecagonal numbers are:" << endl;
		for (i = 1;i <= 100;i++)
		{
			cout << p[i] << " ";
		}
		break;
	case 15:
		p = Pentadecagonal_numbers();
		cout << "The first 100 pentadecagonal numbers are:" << endl;
		for (i = 1;i <= 100;i++)
		{
			cout << p[i] << " ";
		}
		break;
	case 16:
		p = Hexadecagonal_numbers();
		cout << "The first 100 hexadecagonal numbers are:" << endl;
		for (i = 1;i <= 100;i++)
		{
			cout << p[i] << " ";
		}
		break;
	case 17:
		p = Heptadecagonal_numbers();
		cout << "The first 100 heptadecagonal numbers are:" << endl;
		for (i = 1;i <= 100;i++)
		{
			cout << p[i] << " ";
		}
		break;
	case 18:
		p = Octadecagonal_numbers();
		cout << "The first 100 octadecagonal numbers are:" << endl;
		for (i = 1;i <= 100;i++)
		{
			cout << p[i] << " ";
		}
		break;
	case 19:
		p = Enneadecagonal_numbers();
		cout << "The first 100 enneadecagonal numbers are:" << endl;
		for (i = 1;i <= 100;i++)
		{
			cout << p[i] << " ";
		}
		break;
	case 20:
		p = Icosagonal_numbers();
		cout << "The first 100 icosagonal numbers are:" << endl;
		for (i = 1;i <= 100;i++)
		{
			cout << p[i] << " ";
		}
		break;
	case 21:
		p = Icosihenagonal_numbers();
		cout << "The first 100 icosihenagonal numbers are:" << endl;
		for (i = 1;i <= 100;i++)
		{
			cout << p[i] << " ";
		}
		break;
	case 22:
		p = Icosidigonal_numbers();
		cout << "The first 100 icosidigonal numbers are:" << endl;
		for (i = 1;i <= 100;i++)
		{
			cout << p[i] << " ";
		}
		break;
	case 23:
		p = Icositrigonal_numbers();
		cout << "The first 100 icositrigonal numbers are:" << endl;
		for (i = 1;i <= 100;i++)
		{
			cout << p[i] << " ";
		}
		break;
	case 24:
		p = Icositetragonal_numbers();
		cout << "The first 100 icositetragonal numbers are:" << endl;
		for (i = 1;i <= 100;i++)
		{
			cout << p[i] << " ";
		}
		break;

	case 25:
		cout << "Enter the number of sides for the polygon (between 4 and 10000).\n";
		p = Polygonal_numbers();
		for (i = 1;i <= 100;i++)
		{
			cout << p[i] << " ";
		}
		break;
	default:
		cout << "The polygon must have between 3 and 10000 sides.\n";
	}
	
}
int *Triangular_numbers()
{
	static int triangular_nums[101];
	for (int n = 1; n <= 100; n++)
	{
		int triangular = (n * n + n) / 2;
		triangular_nums[n] = triangular;
	}
	return triangular_nums;
}

int *Square_numbers()
{
	static int square_nums[101];
	for (int n = 1; n <= 100; n++)
	{
		int square = n * n;
		square_nums[n] = square;
	}
	return square_nums;
}
int *Pentagonal_numbers()
{
	static int pentagonal_nums[101];
	for (int n = 1; n <= 100; n++)
	{
		int pentagonal = (3 * n * n - n) / 2;
		pentagonal_nums[n] = pentagonal;
	}
	return pentagonal_nums;
}
int *Hexagonal_numbers()
{
	static int hexagonal_nums[101];
	for (int n = 1; n <= 100; n++)
	{
		int hexagonal = (4 * n * n - 2*n) / 2;
		hexagonal_nums[n] = hexagonal;
	}
	return hexagonal_nums;
}
int *Heptagonal_numbers()
{
	static int heptagonal_nums[101];
	for (int n = 1;n <= 100;n++)
	{
		int heptagonal = (5 * n * n - 3 * n) / 2;
		heptagonal_nums[n] = heptagonal;
	}
	return heptagonal_nums;
}
int* Octagonal_numbers()
{
	static int octagonal_nums[101];
	for (int n = 1;n <= 100;n++)
	{
		int heptagonal = (6 * n * n - 4 * n) / 2;
		octagonal_nums[n] = heptagonal;
	}
	return octagonal_nums;
}
int* Nonagonal_numbers()
{
	static int nonagonal_nums[101];
	for (int n = 1;n <= 100;n++)
	{
		int nonagonal = (7 * n * n - 5 * n) / 2;
		nonagonal_nums[n] = nonagonal;
	}
	return nonagonal_nums;
}
int* Decagonal_numbers()
{
	static int decagonal_nums[101];
	for (int n = 1;n <= 100;n++)
	{
		int decagonal = (8 * n * n - 6 * n) / 2;
		decagonal_nums[n] = decagonal;
	}
	return decagonal_nums;
}
int* Hendecagonal_numbers()
{
	static int hendecagonal_nums[101];
	for (int n = 1;n <= 100;n++)
	{
		int hendecagonal = (9 * n * n - 7 * n) / 2;
		hendecagonal_nums[n] = hendecagonal;
	}
	return hendecagonal_nums;
}
int* Dodecagonal_numbers()
{
	static int dodecagonal_nums[101];
	for (int n = 1;n <= 100;n++)
	{
		int dodecagonal = (10 * n * n - 8 * n) / 2;
		dodecagonal_nums[n] = dodecagonal;
	}
	return dodecagonal_nums;
}
int* Tridecagonal_numbers()
{
	static int tridecagonal_nums[101];
	for (int n = 1;n <= 100;n++)
	{
		int tridecagonal = (11 * n * n - 9 * n) / 2;
		tridecagonal_nums[n] = tridecagonal;
	}
	return tridecagonal_nums;
}
int* Tetradecagonal_numbers()
{
	static int tetradecagonal_nums[101];
	for (int n = 1;n <= 100;n++)
	{
		int tetradecagonal = (12 * n * n - 10 * n) / 2;
		tetradecagonal_nums[n] = tetradecagonal;
	}
	return tetradecagonal_nums;
}
int* Pentadecagonal_numbers()
{
	static int pentadecagonal_nums[101];
	for (int n = 1;n <= 100;n++)
	{
		int pentadecagonal = (13 * n * n - 11 * n) / 2;
		pentadecagonal_nums[n] = pentadecagonal;
	}
	return pentadecagonal_nums;
}
int* Hexadecagonal_numbers()
{
	static int hexadecagonal_nums[101];
	for (int n = 1;n <= 100;n++)
	{
		int hexadecagonal = (14 * n * n - 12 * n) / 2;
		hexadecagonal_nums[n] = hexadecagonal;
	}
	return hexadecagonal_nums;
}
int* Heptadecagonal_numbers()
{
	static int heptadecagonal_nums[101];
	for (int n = 1;n <= 100;n++)
	{
		int heptadecagonal = (15 * n * n - 13 * n) / 2;
		heptadecagonal_nums[n] = heptadecagonal;
	}
	return heptadecagonal_nums;
}
int* Octadecagonal_numbers()
{
	static int octadecagonal_nums[101];
	for (int n = 1;n <= 100;n++)
	{
		int octadecagonal = (16 * n * n - 14 * n) / 2;
		octadecagonal_nums[n] = octadecagonal;
	}
	return octadecagonal_nums;
}
int* Enneadecagonal_numbers()
{
	static int enneadecagonal_nums[101];
	for (int n = 1;n <= 100;n++)
	{
		int enneadecagonal = (17 * n * n - 15 * n) / 2;
		enneadecagonal_nums[n] = enneadecagonal;
	}
	return enneadecagonal_nums;
}
int* Icosagonal_numbers()
{
	static int icosagonal_nums[101];
	for (int n = 1;n <= 100;n++)
	{
		int icosagonal = (18 * n * n - 16 * n) / 2;
		icosagonal_nums[n] = icosagonal;
	}
	return icosagonal_nums;
}
int* Icosihenagonal_numbers()
{
	static int icosihenagonal_nums[101];
	for (int n = 1;n <= 100;n++)
	{
		int icosihenagonal = (19 * n * n - 17 * n) / 2;
		icosihenagonal_nums[n] = icosihenagonal;
	}
	return icosihenagonal_nums;
}
int* Icosidigonal_numbers()
{
	static int icosidigonal_nums[101];
	for (int n = 1;n <= 100;n++)
	{
		int icosidigonal = (20 * n * n - 18 * n) / 2;
		icosidigonal_nums[n] = icosidigonal;
	}
	return icosidigonal_nums;
}
int* Icositrigonal_numbers()
{
	static int icositrigonal_nums[101];
	for (int n = 1;n <= 100;n++)
	{
		int icositrigonal = (21 * n * n - 19 * n) / 2;
		icositrigonal_nums[n] = icositrigonal;
	}
	return icositrigonal_nums;
}
int* Icositetragonal_numbers()
{
	static int icositetragonal_nums[101];
	for (int n = 1;n <= 100;n++)
	{
		int icositetragonal = (22 * n * n - 20 * n) / 2;
		icositetragonal_nums[n] = icositetragonal;
	}
	return icositetragonal_nums;
}

int* Polygonal_numbers()
{
	static int polygonal_nums[101];
	static int s;
	cin >> s;
	if (s > 3 && s <= 10000)
	{
		cout << "The first 100 polygonal numbers with " << s << " sides are:" << endl;
		for (int n = 1; n <= 100; n++)
		{
			int polygonal = ((s - 2) * n * n - (s - 4) * n) / 2;
			polygonal_nums[n] = polygonal;
		}
		return polygonal_nums;
	}
	
}



